### PR TITLE
ci: only enable sdk dev engine checks on main

### PR DESCRIPTION
**Pull request by jedevc**
Based on https://github.com/dagger/dagger/pull/7483

Rework of https://github.com/dagger/dagger/pull/7628, that doesn't entirely disable these checks.

### CODE REVIEW

File: .github/workflows/sdk.yml [hunk 1]
```diff
@@ -22,36 +22,66 @@ concurrency:
 
 jobs:
   go:
+    uses: ./.github/workflows/_sdk_check.yml
+    with:
+      sdk: go
+  go-dev:
```
Comment (matipan) [inline .github/workflows/sdk.yml:28]: When running on main, do we still want to run both `<sdk>-dev` and `<sdk>` jobs?

**Comment by gerhard** [comment]
I would flip the condition & run `dev` in PRs, and stable on `main`. Since PRs are less stable than `main`, I would think that testing `dev` is more important there. As for stable, that seems to fit more in `main` since it's checking for backwards compatibility.

The more I think about this, the more I believe that we should run against both `dev` and `stable` all the time. The mental model that I have is:
- did any of the changes in this PR break anything (a.k.a. `dev`) ?
- are the changes in this PR compatible with latest release (a.k.a. `stable`) ?

Since new PRs could be merged in `main` while this PR is open, we want to run the same there since we want to be confident that the combination & ordering of PRs still results in green pipelines.

**Comment by jedevc** [comment]
> I would flip the condition & run `dev` in PRs, and stable on `main`. Since PRs are less stable than `main`, I would think that testing `dev` is more important there. As for stable, that seems to fit more in `main` since it's checking for backwards compatibility.

SGTM - do you want to handle rebasing / switching the check? I can also do it.

**Comment by jedevc** [comment]
I would still like to take this as-is - we should be testing against both dev and non-dev for these jobs, which aren't currently doing at all (we're only using dev).

I think we should rework the priority of how we test `dev` at all, moving towards testing everything on `dev` but still keeping around a few more dedicated compatibility checks - but I think that's out of scope for here.

**Comment by gerhard** [comment]
Looking at this now.

**Review by gerhard — APPROVED** [review]
👍 🚀

**Pull request merged**